packages/cli-platform-apple/src/commands/runCommand/getBuildPath.ts

```diff
@@ -24,11 +25,30 @@ export async function getBuildPath(
     throw new CLIError('Failed to get product name.');
   }
 
+  // Default is iOS, the most common build
+  let buildPath = path.join(targetBuildDir, executableFolderPath);
+
+  // catalyst builds need a suffix, but this regresses from time to time
+  // so check - there may be one already, or we may need to add suffix
   if (isCatalyst) {
-    return path.join(`${targetBuildDir}-maccatalyst`, executableFolderPath);
-  } else if (platform === 'macos') {
-    return path.join(targetBuildDir, fullProductName);
-  } else {
-    return path.join(targetBuildDir, executableFolderPath);
+    // make sure path has one and only one '-maccatalyst' on end
+    if (!targetBuildDir.match(/-maccatalyst$/)) {
+      targetBuildDir = `${targetBuildDir}-maccatalyst`;
+    }
+    buildPath = path.join(targetBuildDir, executableFolderPath);
+  }
+
+  // macOS gets the product name, not the executable folder path
+  if (platform === 'macos') {
```

Comment:
shouldn't this be `if-else if`?

Reply:
it certainly could be, but...it seemed to me the three cases were all mutually exclusive so simple `if` overrides/clobbers were semantically identical ?

case 1 - we're the default platform == ios && isCatalyst == false:  default case join targetBuildDir + executableFolderPath

case 2 - we're the rare platform == ios && isCatalyst == true: clobber with result of maybe adding `-maccatalyst` to targetBuildDir  -- **note that catalyst is an ios platform subset, it will not go in to the platform == macos conditional as true**

case 3 - we're the platform == macos: clobber join of targetBuildDir + fullProductName

Each of the three cases is unambiguously identifiable and unique so I thought a simple set of ifs + clobbers worked?

Reply:
Yes, they were equivalent. But with 
```
if <cond1> {
}
if <cond2> {
}
if <cond3> {
}
```
All three conditions are always evaluated. With 

```
if <cond1> {
} else if <cond2> {
} else if <cond3> {
}
```
cond2 is evaluated if and only if cond1 is false. cond3 is evaluated if and only if both cond1 and cond2 are false.

So, the if-elseif approach is slightly more efficient, and makes it clear that the conditions are mutually exclusive.

Reply:
@cipolleschi lots of discussion and we are in agreement with each other :-) - the question is that with the mixed signal I am hearing of "if/elseif/elseif is slightly better" vs "but I have approved the if/if/if version" - do you want me to convert it and re-push or not?

I don't want to swirl endlessly on this - it's a relatively small fix after all, but I'll do what you like (because, it's an easy conversion from if/if/if/ to if/elseif/elsif too).

So just let me know but I won't change+repush unless you ask, so we don't have another approval cycle unless desired

Cheers man